bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb

```diff
@@ -130,6 +130,67 @@
         end
       end
 
+      context "with a Bundler v2 version specified", :bundler_v2_only do
+        let(:requirement_string) { "~> 1.4.0" }
+
+        let(:dependency_files) { project_dependency_files("bundler2/bundler_specified") }
+        its([:version]) { is_expected.to eq(Gem::Version.new("1.4.0")) }
+
+        context "attempting to update Bundler" do
+          let(:dependency_name) { "bundler" }
+          let(:requirement_string) { "~> 2.2.0" }
+
+          let(:dependency_files) { project_dependency_files("bundler2/bundler_specified") }
+
+          it "returns nil as resolution returns the bundler version installed by core" do
+            expect(subject).to be_nil
+          end
+        end
+      end
+
+      context "with a dependency that requiers bundler v1", :bundler_v1_only do
+        let(:dependency_name) { "guard-bundler" }
+        let(:requirement_string) { "2.2.1" }
+
+        let(:dependency_files) { project_dependency_files("bundler1/requires_bundler") }
+        its([:version]) { is_expected.to eq(Gem::Version.new("2.2.1")) }
+      end
+
+      context "when bundled with v1 and requesting a version that requires bundler v2", :bundler_v1_only do
+        let(:dependency_name) { "guard-bundler" }
+        let(:requirement_string) { "~> 3.0.0" }
+
+        let(:dependency_files) { project_dependency_files("bundler1/requires_bundler") }
+
+        it "raises a DependencyFileNotResolvable error" do
+          expect { subject }.
+            to raise_error(Dependabot::DependencyFileNotResolvable)
+        end
+      end
+
+      context "with a dependency that requiers bundler v2", :bundler_v2_only do
+        let(:dependency_name) { "guard-bundler" }
+        let(:requirement_string) { "3.0.0" }
+
+        let(:dependency_files) { project_dependency_files("bundler2/requires_bundler") }
+
+        pending "resolves version" do
+          expect(subject.version).to eq(Gem::Version.new("3.0.0"))
+        end
+      end
+
+      context "when bundled with v2 and requesting a version that requires bundler v1", :bundler_v2_only do
```

Comment:
@jurre looks like this spec is not actually raising, not sure there's anything wrong with that though so probably just a bogus test case that can be removed.